File: 3432_Count_Partitions_with_Even_Sum_Difference/solution.py
class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        total_sum, left_sum, count = sum(nums), 0, 0
        for i in range(len(nums) - 1):
            left_sum += nums[i]
            right_sum = total_sum - left_sum
            if (left_sum - right_sum) % 2 == 0: count += 1
            
        return count

File: 3432_Count_Partitions_with_Even_Sum_Difference/solution.c
int countPartitions(int* nums, int numsSize) {
    int t_sum = 0, l_sum = 0, count = 0;
    for (int i = 0; i < numsSize; i++) t_sum += nums[i];
    for (int i = 0; i < numsSize - 1; i++) {
        l_sum += nums[i];
        int r_sum = t_sum - l_sum;
        if ((l_sum - r_sum) % 2 == 0) count++;
    }
    return count;
}